pkg/controllers/proxy/network_services_controller.go

```diff
@@ -793,6 +793,42 @@ func (nsc *NetworkServicesController) publishMetrics(serviceInfoMap serviceInfoM
 	return nil
 }
 
+func epCompatible(a, b []endpointsInfo) bool {
```

Comment:
I would name these functions something different here that is more descriptive of what they do. Maybe `comparable` or `equivalent`? I'd look around in the Golang API and see what these types of functions are usually called.

Additionally, it would probably be good to move these functions into a utils file or something. I know that you're just following what's already been done, but these controllers are getting really lengthy and its hard to follow the logic. So we should probably start splitting simple functions like this out.

Reply:
I've not used Go much, but wouldn't such a method have to - by definition - be a public one in order to be extracted to a separate utility file?
Which would be a bit strange as it's a rather single-purpose method to check if two endpoint maps inside the update are compatible.

Reply:
Yes, it would have to be public, and while its a single use now, this seems like the type of thing that will doubtless be useful later. https://github.com/cloudnativelabs/kube-router/blob/master/pkg/utils/service.go would be a good spot to put this.

Reply:
As I'm not very used to Go, what would be the best way to set this up, as the `endpointInfo` structs can't be used for an `interface{}` method - if I want to put it as a generic method.
Considering the data in question is marked as internal, should it be moved to be public as well?